### PR TITLE
Schedule E2E Tests to run every 6 hours instead of every PR

### LINKED ISSUE #3290: Run E2E Tests on schedule
Currently, the E2E suite runs on every push for PRs which makes the dev workflow slower. Instead, run E2E tests on schedule (say every 6 hours) and on release branches rather than running it on every PR.

### PR DESCRIPTION
**Pull request by achyuthajoy**
Fixes #3290 

#### Changes proposed in this Pull Request
This PR moves the E2E Tests outside PR's except release branches making the development workflow faster. The tests are scheduled to run every 6 hours.

Discussion here - paJDYF-2PM-p2#comment-8636

#### How to Test

- [ ] Confirm that the tests are running every 6 hours. It is possible that there might be a slight delay according to Github docs - `The schedule event can be delayed during periods of high loads of GitHub Actions workflow runs.`

-------------------

- [ ] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 name: E2E tests
 
 on:
+  schedule:
+    - cron: '0 */6 * * *'
   pull_request:
     branches:
-      - trunk
-      - develop
       - release/*
```
Comment (RadoslavGeorgiev) [inline .github/workflows/e2e-test.yml:8]: Reading https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestbranchestags,I see the following:

> For a `pull_request` event, only branches and tags on the base are evaluated.

Based on this line, and the fact that the workflow worked on pull requests to `trunk` and `develop`, I understand that `on.pull_request.<branch>` works based on the branch you want to merge the PR onto.

With this in mind, I'd suggest keeping the `trunk` line to run tests on PRs, which are about to get merged into `trunk`. This includes release PRs, as during deployment, they get merged to `trunk`.

I'm thinking whether we should keep `release/*` here, as generally PRs tend to be merged into `develop`, or `trunk` during patch releases. `release/*` branches are just temporary branches, used to verify the contents of a release shortly before merging them to `trunk`, and don't live long enough for PRs to be opened on top of them.

**Review by RadoslavGeorgiev — APPROVED** [review]
I read the P2 discussion, and the schedule seems 👌, but I'm not sure if the branches are correct. Could you please verify if I understood the action correctly?

**Review by RadoslavGeorgiev — CHANGES_REQUESTED** [review]
Approved by mistake, I wanted to request changes.

**Review by RadoslavGeorgiev — APPROVED** [review]
Looks good, thank you @achyuthajoy! 🥳 

:shipit:!

**Comment by aprea** [comment]
@achyuthajoy @RadoslavGeorgiev I'm a little confused about the functionality of this PR.

This is my understanding of how the changes in this PR will work:
- E2E tests are run periodically every 6 hours on PRs that target `trunk`.

Is this correct?

If so, I'm not sure I understand this approach.

Given the existing release train process, we typically only open PRs to `trunk` on release day (Wednesday) on the 3rd week of the cycle.

Does this mean we will only run E2E tests on release day when the release branch/PR is opened?

This doesn't seem ideal. It means we'll have 2 weeks of branches being merged into `develop` without knowing if they break the E2E tests.

Apologies if I have missed something here or made incorrect assumptions.

**Comment by achyuthajoy** [comment]
Hey @aprea The E2E tests will run every 6 hours + it will run on every PR to trunk.

However your reaponse made me think about the branch on which the test runs every 6 hours. I'll check it out and update the PR if required.

Thanks for chiming in 😉

**Comment by achyuthajoy** [comment]
@aprea Based on GH Actions documentation, the scheduled action will run on the latest commit on the default or base branch.

So yes, the tests will run every 6 hours on `develop` and on every PR to `trunk`.

Hope this clears the confusion.

**Comment by aprea** [comment]
That’s good to know, thanks!

**Comment by jrodger** [comment]
> So yes, the tests will run every 6 hours on develop and on every PR to trunk.
>
> Hope this clears the confusion.

Came here to ask the same thing, thanks for clarifying.

**Pull request merged**